src/ast/ast_validations.c

```diff
@@ -1705,3 +1688,39 @@ AST_Validation AST_Validate(RedisModuleCtx *ctx, const cypher_parse_result_t *re
 	return res;
 }
 
+AST_Validation AST_Validate_QueryParams(RedisModuleCtx *ctx, const cypher_parse_result_t *result) {
+	// Check for failures in libcypher-parser
+	if(AST_ContainsErrors(result)) {
+		char *errMsg = _AST_ReportErrors(result);
+		RedisModule_Log(ctx, "debug", "Error parsing query: %s", errMsg);
+		RedisModule_ReplyWithError(ctx, errMsg);
+		free(errMsg);
+		return AST_INVALID;
+	}
+
+	const cypher_astnode_t *root = cypher_parse_result_get_root(result, 0);
+	// Check for empty query
+	if(root == NULL) {
+		RedisModule_ReplyWithError(ctx, "Error: empty query.");
+		return AST_INVALID;
+	}
+
+	char *reason;
+	cypher_astnode_type_t root_type = cypher_astnode_type(root);
+	if(root_type != CYPHER_AST_STATEMENT) {
+		// This should be unnecessary, as we're currently parsing
+		// with the CYPHER_PARSE_ONLY_STATEMENTS flag.
+		asprintf(&reason, "Encountered unsupported query type '%s'", cypher_astnode_typestr(root_type));
+		RedisModule_ReplyWithError(ctx, reason);
+		free(reason);
+		return AST_INVALID;
+	}
+
+	if(_ValidateInputParameters(root, &reason) != AST_VALID) {
```

Comment:
This call (really, this whole function) is invoked whether or not we have parameters, which seems wasteful to me. We can at least skip this one if `cypher_ast_statement_noptions(root) == 0`, but it would be preferable to not get here at all if possible.